mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -475,19 +478,46 @@ class ExperimentView extends Component {
       });
   }
 
-  onCheckbox(runUuid) {
-    const newState = Object.assign({}, this.state);
-    if (this.state.runsSelected[runUuid]) {
-      delete newState.runsSelected[runUuid];
-      this.setState(newState);
-    } else {
-      this.setState({
-        runsSelected: {
-          ...this.state.runsSelected,
-          [runUuid]: true,
+  /**
+   * Handler for a click event on a checkbox in the runs table. Handles both clicking individual
+   * runs and shift-clicking to select or deselect multiple contiguous runs.
+   * @param event Click event
+   * @param childrenIds Child run UUIDs of the current run
+   * @param index Index of the current run within sortedRunIds.
+   * @param sortedRunIds List of run UUIDs (both visible and collapsed) sorted by the current
+   *                     display order.
+   */
+  onCheckbox(event, childrenIds, index, sortedRunIds) {
```

Comment:
Btw, this does seem a bit odd, how much state has to be passed through the `onCheckbox` handler. It seems like onCheckbox should receive the minimal amount of information to make a decision -- e.g., an index in the table. It could then observe the state of the world to make decisions about what to do. Otherwise we're just passing around all the state to the callback, and if we need more state we'll have to add that too.

Reply:
Agree that it's weird. I might defer the refactor needed to expose some of the arguments here (`childrenIds, sortedRunIds`) as higher-level "state of the world" for now, but in the future we could:

* Compute childrenIds per run when we receive a SearchRuns API response / store this info in the Redux store so that it's globally available within our application
* Maintain sorted run IDs as a state attribute of ExperimentView, update it when sort handlers are clicked.

Some initial work towards such a refactor is visible in https://github.com/smurching/mlflow/pull/1/files (for future reference)